client/src/Backoffice/components/Main/components/modals/AddRide/NewRideInfo/NewRideInfo.tsx

```diff
@@ -157,6 +208,38 @@ function NewRideInfo() {
             }}
           />
         </FormControl>
+        <FormControl>
+          <TimePicker
+            sx={{ width: '100%' }}
+            label="שעת איסוף"
+            disablePast
+            ampm={false}
+            value={timeInIsrael}
+            onChange={(time) => {
+              if (time) {
+                const existingDate = watch().pickupDateTime || dayjs();
+                let newDateTime;
+                if (existingDate instanceof Date) {
+                  newDateTime = new Date(existingDate);
+                } else {
+                  newDateTime = dayjs(existingDate).toDate();
```

Comment:
@aviv-taori What is this trying to achieve? In case `existingDate` is not a `Date` what else can it be?